src/Integration/Telemetry/TelemetryDataRepository.cs

```diff
@@ -86,53 +84,49 @@ private void OnStorageFileChanged(object sender, FileSystemEventArgs e)
         private void EnsureFileExists(string filePath)
         {
             var directoryPath = Path.GetDirectoryName(filePath);
-            if (!directoryWrapper.Exists(directoryPath))
+            if (!fileSystem.Directory.Exists(directoryPath))
             {
-                directoryWrapper.Create(directoryPath);
+                fileSystem.Directory.CreateDirectory(directoryPath);
             }
-            if (!fileWrapper.Exists(filePath))
+            if (!fileSystem.File.Exists(filePath))
             {
                 WriteToXmlFile();
             }
         }
 
         private void WriteToXmlFile()
         {
-            TextWriter writer = null;
             var success = RetryHelper.RetryOnException(3, TimeSpan.FromSeconds(2),
                 () =>
                 {
-                    writer = fileWrapper.CreateText(StorageFilePath);
-                    this.ignoreFileChange = true;
-                    telemetrySerializer.Serialize(writer, Data);
-                    writer?.Flush();
+                    ignoreFileChange = true;
```

Comment:
I had to re-write those methods, since File.CreateText returns StreamWriter (in system.io.abstraction's implementation), which is not convertible to StringWriter that we use in the tests.
It used to work in the past because in our FileWrapper we casted it to TextWriter.
I tried to mock StreamWriter using MemoryStream but it was too complicated for the test flow, and File.WriteAllText is easier in my opinion anyway.
I can instead re-write the tests and use System.IO.Abstractions.TestingHelpers.MockFileStream, whatever you prefer.